dev-docs/RFCs/v5.3/generic-accessor-props-rfc.md

```diff
@@ -0,0 +1,89 @@
+# Generic Accessors Micro-RFC
+
+## Background
+
+Many layers use accessors that return constants, for example:
+
+```js
+new ScatterplotLayer({
+  radiusScale: 100,
+  getPosition: d => d.position,
+  getRadius: d => 1,
+  getColor: d => [255, 200, 0]
+});
+```
+
+The curent implementation to handle this use case has several problems:
+
+* A large buffer is created and filled with the same value. This can be much more efficient if we switch to using generic attribute.
+* Update triggers are required to change these constants, which has been a pain point for beginners.
+
+
+## Proposal
+
+### Option 1: New Uniform Props
+
+* Add a new value prop for every accessor prop, e.g.
+  + `position` (Number[3]) - corresponds to `getPosition`
+  + `radius` (Number) - corresponds to `getRadius`
+  + `color` (Number[4]) - corresponds to `getColor`
+* The value prop takes precedence over the accessor, i.e. if `radius` is provided, `getRadius` is ignored. This is so that layers are backward compatible.
+
+The above example now becomes:
+
+```js
+new ScatterplotLayer({
+  radiusScale: 100,
+  getPosition: d => d.position,
+  radius: 1,
+  color: [255, 200, 0]
+});
+```
+
+Core changes:
+
+- `LayerManager` checks for value overrides of the specified accessors. The values are automatically applied as generic attributes if provided.
+- Some composite layers need to update prop forwarding to include the new props.
+
+
+### Option 2: Overloaded Accessors
```

Comment:
My preference as you know is to keep the number of props limited, so I prefer option 2.

Reply:
We should check with @jckr etc to make sure we understand the situation, but I believe that the dual props could have historical reasons. react-vis didn't have accessors, and then recently made a push to implement accessors specifically to align with deck.gl. But they couldn't remove the old props without making a major version bump.